shared/Microsoft.AspNetCore.RangeHelper.Sources/RangeHelper.cs

```diff
@@ -21,15 +22,23 @@ internal static class RangeHelper
         /// </summary>
         /// <param name="context">The <see cref="HttpContext"/> associated with the request.</param>
         /// <param name="requestHeaders">The <see cref="RequestHeaders"/> associated with the given <paramref name="context"/>.</param>
+        /// <param name="length">The total length of the file representation requested.</param>
         /// <param name="lastModified">The <see cref="DateTimeOffset"/> representation of the last modified date of the file.</param>
         /// <param name="etag">The <see cref="EntityTagHeaderValue"/> provided in the <see cref="HttpContext.Request"/>.</param>
-        /// <returns>A collection of <see cref="RangeItemHeaderValue"/> containing the ranges parsed from the <paramref name="requestHeaders"/>.</returns>
-        public static ICollection<RangeItemHeaderValue> ParseRange(HttpContext context, RequestHeaders requestHeaders, DateTimeOffset? lastModified = null, EntityTagHeaderValue etag = null)
+        /// <returns>A boolean value which represents if the <paramref name="requestHeaders"/> contain a single valid 
+        /// range request (true), or not (false). A <see cref="RangeItemHeaderValue"/> which represents the normalized form of the 
```

Comment:
```
        /// <returns>A boolean value which represents if the <paramref name="requestHeaders"/> contain a single valid 
        /// range request. A <see cref="RangeItemHeaderValue"/> which represents the normalized form of the 
        /// range parsed from the <paramref name="requestHeaders"/> or <c>null</c> if it cannot be normalized.</returns>
        /// <remark>If the Range header exists but cannot be parsed correctly, or if the provided length is 0, then the range request cannot be satisfied (status 416). 
        /// This results in (<c>true</c>,<c>null</c>) return values.</remark>
```

That said, let's take a closer look at this design. It's weird that we ignore otherwise valid multi-value headers (false, null), but consider some single value headers as not satisfiable (true, null).